src/java/org/apache/cassandra/config/YamlConfigurationLoader.java

```diff
@@ -363,10 +365,12 @@ else if (result.getAnnotation(Deprecated.class) != null)
 
             return new ForwardingProperty(result.getName(), result)
             {
+                boolean allowsNull = result.getAnnotation(Nullable.class) != null;
+
                 @Override
                 public void set(Object object, Object value) throws Exception
                 {
-                    if (value == null && get(object) != null)
+                    if (value == null && get(object) != null && !allowsNull)
```

Comment:
It seems inconsistent to only require an `Nullable` annotation for configuration that can be set null, but is not defaulted to setting null.  What do you think about dropping the `get(object)` check, and applying `@Nullable` to all cases in Config that can be legitimately set to `null`?

Reply:
I thought about that yesterday but my concern is that there are too many of those and they are not only from the new types and not only new ones...
Maybe we can open a follow up ticket only on trunk as an improvement? And add more testing, etc... I am afraid we are getting into another rabbit hole.

Reply:
Understand about rabbit holes. I'd be happy with opening a follow up JIRA and adding a comment referencing it in this method.

Reply:
Ticket opened, I linked it to this ticket and added a TODO comment